rdflib/plugins/sparql/optimizer.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+
+"""
+This contains standard optimizers for sparql
+
+"""
+import re
+from rdflib import Literal
+from rdflib.plugins.sparql.operators import Builtin_CONTAINS, Builtin_REGEX
+from rdflib.plugins.sparql.sparql import Query
+from rdflib.plugins.sparql.algebra import CompValue, Join, Values, Expr
+from typing import Any
+
+"""
+An interface for having optimizers that transform a query algebra hopefully
+in an faster to evaluate version.
+"""
+
+
+class SPARQLOptimizer:
+    def optimize(self, query: Query) -> Query:
+        return query
+
+
+class ValuesToTheLeftOfTheJoin(SPARQLOptimizer):
+
+    def optimize(self, query: Query) -> Query:
```

Comment:
```suggestion
    @classmethod
    def optimize(cls, query: Query) -> Query:
```

As these methods don't use the class state and are side effect free it is best to make them class methods, that way it is clearer to users that they don't have to be concerned with concurrency issues.